processor.go

```diff
@@ -34,7 +36,7 @@ type Processor interface {
 	// In case the configuration is not valid it should return an error.
 	// Configure should not open connections or any other resources. It should solely
 	// focus on parsing and validating the configuration itself.
-	Configure(context.Context, map[string]string) error
+	Configure(context.Context, config.Config) error
```

Comment:
Note that I changed the parameter type here, we use the same in the connector SDK and makes it a bit safer to use and know where to use it like in `ParseConfig`.